### ISSUE TITLE
Fix unqualified references to static methods from metadata

### ISSUE THREAD

**Issue by scheglov** [dart-lang/sdk#38641]
In this code `_fromJson` in the metadata is unresolved.
```dart
class DefaultWithToJsonClass {
  @JsonKey(defaultValue: 7, fromJson: _fromJson)
  int fieldDefaultValueToJson;

  static int _fromJson(String input) => 41;
}
```

**Comment by kevmoo** [comment]
Marking P1 – regression in commonly used code!

**Comment by scheglov** [comment]
https://dart-review.googlesource.com/c/sdk/+/119360

**Comment by scheglov** [comment]
The fix for `analyzer` landed in the SDK bleeding edge.

**Comment by kevmoo** [comment]
Could we get a +1 published soonish?

**Comment by scheglov** [comment]
I think I found a bug in my fix :-(
It showed in google3 presubmit.

**Comment by scheglov** [comment]
https://dart-review.googlesource.com/c/sdk/+/119521 looks like it fixes the google3 issue, and the `json_serializable` one too.

**Comment by scheglov** [comment]
Fixed, I think.
Both `json_serializable` and [google3](https://test.corp.google.com/ui#id=OCL:271764619:BASE:272115483:1569898124206:34ce9f52&include_skipped=true) look good.

**Comment by kevmoo** [comment]
Woo hoo! So...uh...could we get a +1 published?

Thanks so much!

**Comment by scheglov** [comment]
@stereotype441 I landed my fixes - this one, and couple others. So, we probably should release a new non-breaking version.

**Comment by kevmoo** [comment]
@stereotype441 ? We have a **BUNCH** of broken packages because of this...

**Comment by stereotype441** [comment]
Ok, I plan to release a new analyzer today.

**Comment by stereotype441** [comment]
Ok, published [analyzer 0.38.5](https://pub.dev/packages/analyzer/versions/0.38.5) including this bug fix.

**Comment by kevmoo** [comment]
Thanks!
